pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -99,8 +99,15 @@ func (b *Builder) BuildHTTP(class networking.ListenerClass) []*hcm.HttpFilter {
 		// Only applies to inbound and gateways
 		return nil
 	}
+	if b.proxy.SupportsEnvoyExtendedJwt() {
+		filter := b.applier.JwtFilter(true, b.proxy.Type != model.SidecarProxy)
+		if filter != nil {
+			return []*hcm.HttpFilter{filter}
+		}
+		return nil
+	}
 	res := []*hcm.HttpFilter{}
-	if filter := b.applier.JwtFilter(); filter != nil {
+	if filter := b.applier.JwtFilter(false, false); filter != nil {
```

Comment:
where's the code to ignore the istio_authn filter. I see we are still building isito_authn filter for all the cases?

Reply:
The above `if` block exits early.

Reply:
ah I missed that. Thanks